Google.PowerShell/BigQuery/BqTable.cs

```diff
@@ -452,51 +451,46 @@ protected override void ProcessRecord()
                     throw UnknownParameterSetException;
             }
 
-            TablesResource.DeleteRequest request = Service.Tables.Delete(Project, DatasetId, Table);
-            String response = "Table Removal Stopped.";
-            if (ShouldProcess(Table))
+            try
             {
-                try
-                {
-                    if (Force)
-                    {
-                        response = request.Execute();
-                    }
-                    else if (noToAll)
-                    {
-                        response = "Removal Stopped: NoToAll";
-                    }
-                    else
-                    {
-                        TablesResource.GetRequest checkTables =
-                            new TablesResource.GetRequest(Service, Project, DatasetId, Table);
-                        var tableResponse = checkTables.Execute();
-                        if (tableResponse.NumRows == 0)
-                        {
-                            response = request.Execute();
-                        }
-                        else if (yesToAll || ShouldContinue(
-                            GetConfirmationMessage(Table, tableResponse.NumRows),
-                            "Confirm Deletion", ref yesToAll, ref noToAll))
-                        {
-                            response = request.Execute();
-                        }
-                    }
-                    WriteObject(response);
-                }
-                catch (GoogleApiException ex) when (ex.HttpStatusCode == HttpStatusCode.NotFound)
+                if (ShouldDelete())
                 {
-                    WriteError(new ErrorRecord(ex,
-                        $"Error {ex.HttpStatusCode}: '{Table}' not found in '{Dataset}'.",
-                        ErrorCategory.ObjectNotFound,
-                        Table));
+                    Service.Tables.Delete(Project, DatasetId, Table).Execute();
                 }
             }
+            catch (GoogleApiException ex) when (ex.HttpStatusCode == HttpStatusCode.NotFound)
+            {
+                WriteError(new ErrorRecord(ex,
+                    $"Error {ex.HttpStatusCode}: '{Table}' not found in '{Dataset}'.",
+                    ErrorCategory.ObjectNotFound, Table));
+            }
+        }
+
+        private bool ShouldDelete()
+        {
+            if (!ShouldProcess(Table) || noToAll)
+            {
+                return false;
+            }
+            else if (Force || yesToAll)
+            {
+                return true;
+            }
+            else
+            {
+                var tableResponse = new TablesResource.GetRequest(
+                    Service, Project, DatasetId, Table).Execute();
+
+                return (tableResponse.NumRows == 0) ? true : 
```

Comment:
`(tableResponse.NumRows == 0) || ShouldContinue` might be a little clearer, but that is debatable.